Forge.TreeWalker/src/ForgeSchemaValidator.cs

```diff
@@ -0,0 +1,263 @@
+
+            return res;
+        }
+
+        private static List<JObject> GetAllSchemasInDirectory(string path, bool validateAsDictionary)
+        {
+            List<JObject> schemaList = new List<JObject>();
+            string[] files = Directory.GetFiles(path);
+            List<object> schemalist = new List<object>();
+
+            if (validateAsDictionary)
+            {
+                Dictionary<string, ForgeTree> combinedDictionary = new Dictionary<string, ForgeTree>();
+
+                foreach (string file in files)
+                {
+                    string schema = File.ReadAllText(file);
+                    Dictionary<string, ForgeTree> schemaDictionary = JsonConvert.DeserializeObject<Dictionary<string, ForgeTree>>(schema);
+
+                    foreach (KeyValuePair<string, ForgeTree> kvp in schemaDictionary)
+                    {
+                        if (kvp.Value.Tree == null)
+                        {
+                            throw new NullReferenceException("Can not deserialize the schema as dictionary in file: " + file);
+                        }
+
+                        combinedDictionary.Add(kvp.Key, kvp.Value);
+                    }
+                }
+
+                return new List<JObject> { SerializeToJObject(combinedDictionary) };
+            }
+            else
+            {
+                foreach (string file in files)
+                {
+                    List<JObject> schemasInFile = GetSchemaFromPath(file, validateAsDictionary);
+                    schemasInFile.ForEach(n => schemaList.Add(n));
+                }
+            }
+
+            return schemaList;
+        }
+
+        /// <summary>
+        /// Validate the input Schemas in JObject against rules in JSchema and return the validation error list. 
+        /// </summary>
+        /// <param name="schemas">One or serveral schemas to be validated. The number of JObjects depends on the input and three cases mentioned in ConvertStringToJObjectList method</param>
+        /// <param name="rules">The rules used to validate input schemas is only allowed in string or JSchema type</param>
+        /// <returns>The result of schema validation. The errorList is the origin error message from IsValid method</returns>
+        private static bool Validate(List<JObject> schemas, object rules, out IList<ValidationError> errorList)
+        {
+            JSchema jSchemaRules = new JSchema();
+            errorList = new List<ValidationError>();
+
+            if (rules is string && !string.IsNullOrWhiteSpace((string)rules))
+            {
+                jSchemaRules = JSchema.Parse((string)rules);
+            }
+            else if (rules is JSchema)
+            {
+                jSchemaRules = (JSchema)rules;
+            }
+            else 
+            {
+                throw new ArgumentException("Rules argument could only be a non-empty string or a JSchema");
```

Comment:
>Rules argument could only be a non-empty string or a JSchema [](http://example.com/codeflow?start=45&length=60)

"Rules argument must be a JSchema or a string that can be parsed into a JSchema."